streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -456,12 +457,14 @@ public void postCommit() {
 
     @Override
     public void closeClean() {
+        streamsMetrics.removeAllTaskLevelSensors(Thread.currentThread().getName(), id.toString());
```

Comment:
We previously relied on the task manager to remove these sensors before calling close, but forgot to do it before recycling. In retrospect, it's better to do it within the same class that creates the sensors to begin with.

Reply:
Agreed, we should clean up anything we created in the same class